test_mavros/sitl_test/lib/sitl_test.cpp

```diff
@@ -0,0 +1,35 @@
+/**
+ * @brief SITL test class
+ * @file sitl_test.cpp
+ * @author Nuno Marques <n.marques21@hotmail.com>
+ */
+/*
+ * Copyright 2015 Nuno Marques.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros.h>
+#include <sitl_test/sitl_test.h>
+#include <sitl_test/test_type.h>
+#include <ros/console.h>
+
+using namespace sitltest;
+using namespace testtype;
+
+SitlTest::SitlTest()
+{}
+
+void SitlTest::spin(int argc, char *argv[])
+{
+	if (strcmp(argv[1],"offboard_control") == 0)
+	{
+		ros::init(argc, argv, "offboard_control");
```

Comment:
I don't think it makes any difference because it's just a node at a time that is launched. And that node is launched based on the args given on `rosrun`/`roslaunch`, so it's a shorter and efficient implementation and didn't give me any problems till now.